dev/SearchBox.js

```diff
@@ -1,6 +1,9 @@
 import React, {Component, PropTypes as T} from 'react';
 import connect from '../src/connect.js';
 
-export default connect()(function({helper}) {
-  return <input onChange={(e) => helper.setQuery(e.target.value).search()} />;
+export default connect()(function({helper, search}) {
```

Comment:
> However, it might break existing code in some edge cases where we rely on it to be synchronous.

When we decide to backport that in the Helper (which would be awesome :D), let's try to figure out what are those edge cases.

> in order to add some default state

Are you implementing default state so that reset still applies the default values? If so, this would be great in the Helper as well :)